activerecord/test/cases/adapters/postgresql/uuid_test.rb

```diff
@@ -403,3 +403,37 @@ def test_find_by_with_uuid
     assert_nil UuidPost.find_by(id: 789)
   end
 end
+
+class PostgresqlUUIDCaseInsensitive < ActiveRecord::PostgreSQLTestCase
+  include PostgresqlUUIDHelper
+
+  class UuidModel < ActiveRecord::Base
+    self.table_name = "pg_uuid_models"
+  end
+
+  setup do
+    connection.transaction do
+      connection.create_table("pg_uuid_models", id: :uuid, **uuid_default) do |t|
+        t.string "title"
+      end
+    end
+  end
+
+  teardown do
+    drop_table "pg_uuid_models"
+  end
+
+  def test_upcase_not_dirty
+    uuid = "abcd-0123-4567-89ef-dead-beef-0101-1010"
+    model = UuidModel.create!(id: uuid)
+    model.id = uuid.swapcase
+    assert_not model.changed?
+  end
+
+  def test_downcase_not_dirty
+    uuid = "ABCD-0123-4567-89EF-DEAD-BEEF-0101-1010"
+    model = UuidModel.create!(id: uuid)
+    model.id = uuid.swapcase
+    assert_not model.changed?
+  end
```

Comment:
Looking at the rest of the file now, I think these tests could be moved into the `PostgresqlUUIDTest` class above, perhaps beneath `test_invalid_uuid_dont_match_to_nil`.  You can use `UUIDType` model that's already defined there too.

Reply:
Test cases combined into a single swapcase, and moved into the section with similar test points.